packages/@aws-cdk/aws-codepipeline-actions/lib/ecr/source-action.ts

```diff
@@ -6,6 +6,26 @@ import { Construct } from '@aws-cdk/core';
 import { Action } from '../action';
 import { sourceArtifactBounds } from '../common';
 
+/**
+ * The CodePipeline variables emitted by the ECR source Action.
+ */
+export interface IEcrSourceVariables {
```

Comment:
Why are these interfaces (`IEcrSourceVariables`) and not structs (`EcrSourceVariables`)?

Reply:
I figured I'd use the `I` prefix, as these types are not really customer-facing (by which I mean, customers don't have a reason to declare variables of these types), so I'd figured I'd leave the names without the `I` prefix (like `EcrSourceVariables`) available if we ever need to also do the stand-alone variables classes.

Reply:
Feels odd and technically has runtime implications in jsii (albeit not dramatic). For example an interface is passed by reference in jsii. 

In case we want to introduce variable “placeholders” in the future we can call them “XxxVariablePlaceholder” or something. This will also be a more suitable name as it will provide a hint for how it’s going to be used.

Reply:
OK. Changed the names to remove the `I` prefix.